spec/core/v2/ics-004-packet-semantics/README.md

```diff
@@ -206,513 +196,111 @@ function channelCapabilityPath(portIdentifier: Identifier, channelIdentifier: Id
 
-Once closed, channels cannot be reopened and identifiers cannot be reused. Identifier reuse is prevented because
-we want to prevent potential replay of previously sent packets. The replay problem is analogous to using sequence
-numbers with signed messages, except where the light client algorithm "signs" the messages (IBC packets), and the replay
-prevention sequence is the combination of port identifier, channel identifier, and packet sequence - hence we cannot
-allow the same port identifier & channel identifier to be reused again with a sequence reset to zero, since this
-might allow packets to be replayed. It would be possible to safely reuse identifiers if timeouts of a particular
-maximum height/time were mandated & tracked, and future specification versions may incorporate this feature.
+The IBC router contains a mapping from a reserved application port and the supported versions of that application as well as a mapping from channelIdentifiers to channels.
 
 ```typescript
-function chanCloseConfirm(
-  portIdentifier: Identifier,
-  channelIdentifier: Identifier,
-  proofInit: CommitmentProof | MultihopProof,
-  proofHeight: Height) {
-    abortTransactionUnless(authenticateCapability(channelCapabilityPath(portIdentifier, channelIdentifier), capability))
-    channel = provableStore.get(channelPath(portIdentifier, channelIdentifier))
-    abortTransactionUnless(channel !== null)
-    abortTransactionUnless(channel.state !== CLOSED)
-    connection = provableStore.get(connectionPath(channel.connectionHops[0]))
-    abortTransactionUnless(connection !== null)
-    abortTransactionUnless(connection.state === OPEN)
-
-    // return hops from counterparty's view
-    counterpartyHops = getCounterPartyHops(proofInit, connection)
-
-    expected = ChannelEnd{CLOSED, channel.order, portIdentifier,
-                          channelIdentifier, counterpartyHops, channel.version}
-
-    if (connectionHops.length > 1) {
-      key = channelPath(counterparty.PortId, counterparty.ChannelId)
-      abortTransactionUnless(connection.verifyMultihopMembership(
-        connection,
-        proofHeight,
-        proofInit,
-        channel.connectionHops,
-        key
-        expected))
-    } else {
-      abortTransactionUnless(connection.verifyChannelState(
-        proofHeight,
-        proofInit,
-        channel.counterpartyPortIdentifier,
-        channel.counterpartyChannelIdentifier,
-        expected
-      ))
-    }
-
-    // write may happen asynchronously in the handler defined in the ICS26 spec
-    // if the channel is closing during an upgrade, 
-    // then we can delete all auxiliary upgrade information
-    provableStore.delete(channelUpgradePath(portIdentifier, channelIdentifier))
-    privateStore.delete(counterpartyUpgradePath(portIdentifier, channelIdentifier))
-
-    channel.state = CLOSED
-    provableStore.set(channelPath(portIdentifier, channelIdentifier), channel)
+type IBCRouter struct {
+    versions: portId -> [Version]
```

Comment:
I don't believe we need this in the router anymore. If the portID doesn't support the version the callback will simply error